system/lib/compiler-rt/stack_ops.s

```diff
@@ -30,3 +32,27 @@ stackAlloc:
   global.set __stack_pointer
   local.get 1
   end_function
+
+emscripten_stack_get_current:
+  .functype emscripten_stack_get_current () -> (i32)
+  global.get __stack_pointer
+  end_function
+
+.functype emscripten_stack_get_end () -> (i32)
+.functype emscripten_stack_get_base () -> (i32)
+.globaltype __stack_end, i32
+__stack_end:
+
+emscripten_stack_get_free:
+  # set __stack_base/__stack_end on first call
+  .functype emscripten_stack_get_free () -> (i32)
+  global.get __stack_end
+  i32.eqz
+  if
+    call emscripten_stack_get_end
+    global.set __stack_end
+  end_if
```

Comment:
Hey, thanks for working on this!

Having this in a .s file is a cool approach. However, this `if `branch makes this suboptimal compared to the solution that was presented in #11162 + WebAssembly/binaryen#2849 that did not need the branch :( (the original solution was just a [single sub](https://github.com/WebAssembly/binaryen/pull/2849/files#diff-8cf6999a742cd6a8d9594c858eddc917R219).

Any change we could get rid of this comparison?

Reply:
Sure, that sounds perfectly reasonable.  Should be trivial change.   I'm OK doing it, or you do you want to?

Reply:
Thanks, that would be much appreciated!